app/services/bing_webmaster.py

```
@@ -0,0 +1,171 @@
+    _HISTORY_WINDOW_DAYS = 365
+
+    def __init__(
+        self,
+        credentials: dict[str, Any],
+        *,
+        client: httpx.Client | None = None,
+        timeout: float = 30.0,
+    ) -> None:
+        api_key = credentials.get("api_key") or credentials.get("key") or credentials.get("token")
+        if not api_key:
+            raise ValueError("Bing credentials must include an API key")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)